### ISSUE TITLE
excessive stunnel logging is filling /var/log partition when using HA lizard (did not happen in 7.6)

### ISSUE THREAD

**Issue by michael-newsrx** [xcp-ng/xcp#510]
I have a three node cluster. Updated via yum update to most recent binaries.

```bash
lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	XCP-ng
Description:	XCP-ng release 8.2.0 (xenenterprise)
Release:	8.2.0
Codename:	xenenterprise
```

On all of the nodes, the /var/log/secure file is being spammed with stunnel entries about connects. This is causing me issues with /var/log filling up in short order. This appears to be caused by the entry ```debug = authpriv.5``` in ```/etc/stunnel/xapi.conf```. Changing this value is not an option as the file is rewritten each type the xapi tools stack restarts.

```bash
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Client certificate not requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server done
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write client key exchange
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write change cipher spec
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server session ticket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read change cipher spec
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: New session callback
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Peer certificate was cached (627 bytes)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Session id: 39CC9E8CB775F8DD5368C3C46249079CB5C9229B691842F1C93C3BF3DDD57AB0
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      1 client connect(s) requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      1 client connect(s) succeeded
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      0 client renegotiation(s) requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      0 session reuse(s)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLS connected: new session negotiated
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLSv1.2 ciphersuite: ECDHE-RSA-AES256-SHA384 (256-bit encryption)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Read socket closed (readsocket)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Sending close_notify alert
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS alert (write): warning: close notify
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: SSL_shutdown successfully sent close_notify alert
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS alert (read): warning: close notify
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLS closed (SSL_read)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Sent socket write shutdown
Jul 27 10:25:11 xen1 stunnel: LOG5[0]: Connection closed: 249 byte(s) sent to TLS, 103 byte(s) sent to socket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Remote descriptor (FD=3) closed
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Service [stunnel] finished (0 left)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Deallocating section defaults
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: Clients allowed=500
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: stunnel 5.56 on x86_64-koji-linux-gnu platform
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: Compiled/running with OpenSSL 1.1.1c FIPS  28 May 2019
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: Threading:PTHREAD Sockets:POLL,IPv6 TLS:ENGINE,FIPS,OCSP,SNI Auth:LIBWRAP
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: errno: (*__errno_location ())
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: Reading configuration from descriptor 7
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: UTF-8 byte order mark not detected
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: FIPS mode disabled
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: No PRNG seeding was required
Jul 27 10:25:11 xen1 stunnel: LOG6[ui]: Initializing inetd mode configuration
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: Ciphers: ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-GCM-SHA384:AES256-SHA256:AES128-SHA256
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: TLSv1.3 ciphersuites: TLS_CHACHA20_POLY1305_SHA256:TLS_AES_256_GCM_SHA384:TLS_AES_128_GCM_SHA256
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: TLS options: 0x02100004 (+0x00000000, -0x00000000)
Jul 27 10:25:11 xen1 stunnel: LOG7[ui]: No certificate or private key specified
Jul 27 10:25:11 xen1 stunnel: LOG4[ui]: Service [stunnel] needs authentication to prevent MITM attacks
Jul 27 10:25:11 xen1 stunnel: LOG5[ui]: Configuration successful
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Service [stunnel] started
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Setting local socket options (FD=0)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Option TCP_NODELAY not supported on local socket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Setting local socket options (FD=1)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Option TCP_NODELAY not supported on local socket
Jul 27 10:25:11 xen1 stunnel: LOG5[0]: Service [stunnel] accepted connection from unnamed socket
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: s_connect: connecting 192.168.1.4:443
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: s_connect: s_poll_wait 192.168.1.4:443: waiting 10 seconds
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: FD=6 events=0x2001 revents=0x0
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: FD=3 events=0x2005 revents=0x0
Jul 27 10:25:11 xen1 stunnel: LOG5[0]: s_connect: connected 192.168.1.4:443
Jul 27 10:25:11 xen1 stunnel: LOG5[0]: Service [stunnel] connected remote server from 192.168.1.3:43250
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Setting remote socket options (FD=3)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Option SO_KEEPALIVE set on remote socket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Option TCP_NODELAY set on remote socket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Remote descriptor (FD=3) initialized
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: SNI: sending servername: 192.168.1.4
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Peer certificate not required
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): before SSL initialization
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write client hello
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write client hello
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server hello
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Certificate verification disabled
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Certificate verification disabled
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server certificate
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server key exchange
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Client certificate not requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server done
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write client key exchange
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write change cipher spec
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS write finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read server session ticket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read change cipher spec
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS state (connect): SSLv3/TLS read finished
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: New session callback
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Peer certificate was cached (627 bytes)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Session id: B7F9F10DA2057F878BCAC038C6B8A4F3A189C93017743AED0444FCA654AEB60B
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      1 client connect(s) requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      1 client connect(s) succeeded
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      0 client renegotiation(s) requested
Jul 27 10:25:11 xen1 stunnel: LOG7[0]:      0 session reuse(s)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLS connected: new session negotiated
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLSv1.2 ciphersuite: ECDHE-RSA-AES256-SHA384 (256-bit encryption)
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: Read socket closed (readsocket)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Sending close_notify alert
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS alert (write): warning: close notify
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: SSL_shutdown successfully sent close_notify alert
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: TLS alert (read): warning: close notify
Jul 27 10:25:11 xen1 stunnel: LOG6[0]: TLS closed (SSL_read)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Sent socket write shutdown
Jul 27 10:25:11 xen1 stunnel: LOG5[0]: Connection closed: 291 byte(s) sent to TLS, 71 byte(s) sent to socket
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Remote descriptor (FD=3) closed
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Service [stunnel] finished (0 left)
Jul 27 10:25:11 xen1 stunnel: LOG7[0]: Deallocating section defaults
```

https://www.halizard.com/forum/suggestion-box/712-excessive-stunnel-logging

**Comment by michael-newsrx** [comment]
Possible fix. Add back file ```/etc/cron.d/logrotate.cron```. It appears to have been removed during an update as I have logrorate.conr.rpmsave files. Therefore the once a minute failsafe make sure /var/log isn't full check isn't run anymore.

Contents:

```bash
*  * * * * root /opt/xensource/bin/logrotate-xenserver
```

**Comment by ecoutinho** [comment]
That might not be it. Although my servers are not fully updated, they also have that logrorate.conf.rpmsave. /var/log/secure starts at 4AM and is only 5MB. You do have many more stunnel messages. Have you tried a xe-toolstack-restart on each node?

**Comment by michael-newsrx** [comment]
yes, full restarts on all nodes.
    
    On 7/27/21 11:12 AM, Elio Coutinho
      wrote:
    
    
      
      That might not be it. Although my servers are not fully
        updated, they also have that logrorate.conf.rpmsave.
        /var/log/secure starts at 4AM and is only 5MB. You do have many
        more stunnel messages. Have you tried a xe-toolstack-restart on
        each node?
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/xcp-ng/xcp/issues/510#issuecomment-887597199",
"url": "https://github.com/xcp-ng/xcp/issues/510#issuecomment-887597199",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by stormi** [comment]
> Possible fix. Add back file `/etc/cron.d/logrotate.cron`. It appears to have been removed during an update as I have logrorate.conf.rpmsave files. Therefore the once a minute failsafe make sure /var/log isn't full check isn't run anymore.
> 
> Contents:
> 
> ```shell
> *  * * * * root /opt/xensource/bin/logrotate-xenserver
> ```

It's not a failed update, it's a (questionable) design choice from Citrix teams to leave a .rpmsave file here so that it's available when they need it. There exist two logging configurations on XenServer and thus XCP-ng: the old, aggressive, log rotation, from the time when there was no log partition and really small root partitions, and the newer, daily, log rotation with 30 days of history.

So now you have two log rotation cron jobs enabled at the same time, each using a different logrotate configuration. It may fare well, but I can't guarantee it.

**Comment by michael-newsrx** [comment]
As of right now this is the only workaround that I now of when running into this issue.

**Comment by stormi** [comment]
It seems it was fixed in the XAPI project: https://github.com/xapi-project/xen-api-libs-transitional/commit/71657bd21f7105e50ab0cf3c47084f0ff651f5c2

**Comment by stormi** [comment]
> As of right now this is the only workaround that I now of when running into this issue.

It would be better to add a cron job to rotate only `secure.log`, temporarily. The old `logrotate-xenserver` command discards a lot of log files, not just `secure.log`.

**Comment by michael-newsrx** [comment]
We are a SOHO setup w/o an admin. Having the files rotated that we
    don't review isn't an issue for us.
    
    On 7/28/21 8:22 AM, Samuel VERSCHELDE
      wrote:
    
    
      
      
        As of right now this is the only workaround that I now of
          when running into this issue.
      
      It would be better to add a cron job to rotate only secure.log,
        temporarily. The old logrotate-xenserver command
        discards a lot of log files, not just secure.log.
      —
        You are receiving this because you authored the thread.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/xcp-ng/xcp/issues/510#issuecomment-888264846",
"url": "https://github.com/xcp-ng/xcp/issues/510#issuecomment-888264846",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by stormi** [comment]
Are your hosts up to date? The latest update to `xapi*` packages should have reduced the verbosity of those logs already.

**Comment by stormi** [comment]
> Are your hosts up to date? The latest update to `xapi*` packages should have reduced the verbosity of those logs already.

My mistake: the update did not contain the fix. However it is queued for the next update that will contain the `xapi*` packages.

**Comment by ethanjosephscott** [comment]
Hey Gang,
I worked on a similar enough setup this week with excessive stunnel logging, I can confirm the newest xapi patch and a restart will resolve this issue.

**Comment by stormi** [comment]
Thanks for the feedback!

**Comment by ethanjosephscott** [comment]
My sincere apologies,  it appears I had checked the log partition of my plain Xen Host and not my XCP-ng host.  Disregard my comment.

**Comment by stormi** [comment]
A fix is now available as an update candidate: https://xcp-ng.org/forum/topic/365/updates-announcements-and-testing/306

Please test it if possible, and also pass the word on the HA lizard forum.

**Comment by stormi** [comment]
The update that fixes this issue has now been released for XCP-ng 8.2: https://xcp-ng.org/blog/2021/12/08/december-2021-xcp-ng-updates/
